File: app/src/main/java/acr/browser/lightning/favicon/FaviconUtils.java
package acr.browser.lightning.favicon;

import android.net.Uri;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.text.TextUtils;

/**
 * Simple utils for favicon fetching.
 */
public class FaviconUtils {
    @Nullable
    public static Uri safeUri(@NonNull String url) {
        if (TextUtils.isEmpty(url)) {
            return null;
        }

        Uri uri = Uri.parse(url);

        if (TextUtils.isEmpty(uri.getScheme()) || TextUtils.isEmpty(uri.getHost())) {
            return null;
        }

        return uri;
    }

    static void assertUriSafe(@Nullable Uri uri) {
        if (uri == null || TextUtils.isEmpty(uri.getScheme()) || TextUtils.isEmpty(uri.getHost())) {
            throw new RuntimeException("Unsafe uri provided");
        }
    }
}

File: app/src/main/java/acr/browser/lightning/favicon/FaviconUtils.kt
@file:JvmName("FaviconUtils")

package acr.browser.lightning.favicon

import android.net.Uri

/**
 * Requires that a [Uri] be safe for use. The [Uri.getScheme] and [Uri.getHost] must not be
 * `null` or blank.
 */
fun requireUriSafe(uri: Uri) {
    if (uri.scheme?.isNotBlank() != true || uri.host?.isNotBlank() != true) {
        throw RuntimeException("Unsafe uri provided")
    }
}
